spec/support/concerns/job_transactionable_shared_examples.rb

```diff
@@ -0,0 +1,58 @@
+shared_examples 'a job_transactionable model' do
+  it {
+    expect(described_class).to include(JobTransactionable)
+    is_expected.to respond_to(:current_transaction)
+    is_expected.to have_many(:job_transactions)
+
+    is_expected.to respond_to(:root_create_transaction)
+    is_expected.to respond_to(:root_update_transaction)
+    is_expected.to callback(:root_create_transaction)
+      .before(:create)
+    is_expected.to callback(:root_update_transaction)
+      .before(:update)
+  }
+
+  describe '.create_transaction' do
+    let(:transaction_state) {'testing'}
+
+    context 'with nil current_transaction' do
+      it {
+        subject.create_transaction(transaction_state)
+        expect(subject.current_transaction).to be
+        expect(subject.current_transaction).not_to be_persisted
+        expect(subject.current_transaction.state).to eq(transaction_state)
+      }
+    end
+
+    context 'with preset current_transaction' do
+      let(:existing_current_transaction)  {
+        subject.create_transaction 'pretest'
+      }
+
+      it {
```

Comment:
It would be good to test the contents of `#current_transaction`. Also, `existing_current_transaction` isn't called, so I'm guessing that the current transaction is actually nil.  :\